src/firefly/js/api/ApiHighlevelBuild.js

```diff
@@ -0,0 +1,227 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+//===================================================================================
+//-----------------------------------------------------------------------------------
+// Build the firefly high level api
+// This file should have has no imports.  It should be build the high level api completely from the lowlevel.
+// There are two reasons for this.
+//   1. It will be an example of how to use the lowlevel api
+//   2. We want to make sure the the lowlevel api is only built with the high level.  That
+//      way we know that the lowlevel is complete.
+//-----------------------------------------------------------------------------------
+//===================================================================================
+
+/**
+ * 
+ * @param llApi the lowlevel api
+ * @return {{}}
+ */
+export function buildHighLevelApi(llApi) {
+    const current= build(llApi);
+    const deprecated= buildDeprecated(llApi);
+
+    return Object.assign({}, deprecated, current);
+}
+
+
+var globalImageViewDefParams= {};
+
+/**
+ * Build the deprecated API
+ * @param llApi
+ * @return {{}}
+ */
+function build(llApi) {
+
+    const imagePart= buildImagePart(llApi);
+    const xyPart= buildXYandHistPart(llApi);
+    const tablePart= buildTablePart(llApi);
+    return Object.assign({}, imagePart,xyPart,tablePart);
+}
+
+
+function buildTablePart(llApi) {
+    //todo
+    return {};
+}
+
+function buildXYandHistPart(llApi) {
+    //todo
+    return {};
+}
+
+
+function buildImagePart(llApi) {
+
+
+    const hApi= {};
+    const {RequestType}= llApi.util.image;
+
+    /**
+     * The general plotting function to plot a FITS image.
+     * @param targetDiv to put the image in.
+     * @param request the request object literal with the plotting parameters
+     */
+    hApi.showImage= (targetDiv, request)  => showImageInMultiViewer(llApi, targetDiv, request);
```

Comment:
these functions are nicely declared and exposed here.  but, it may be hard to generate jsdoc from it.  may need to revisit when we start looking at generating jsdoc.